docs-kits/kits/Digital Twin Kit/Software Development View/page_interaction-patterns.md

```diff
@@ -0,0 +1,134 @@
+    - registers Digital Twins (ShellDescriptors in AAS vocabulary) with specificAssetIds including at least one that's
+      equivalent to one attached to the OEM's twin.
+- Submodel (like [_Certificate of Destruction_](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/4889d8482fc6b233eb0f56f3ac94c5ea4004bc05/io.catenax.certificate_of_destruction/1.0.1/gen/CertificateOfDestruction.json))
+    - registered in a DSP-conformant connector to the Dataspace.
+    - registered to the correct ShellDescriptor in the DTR.
+    - Serving arbitrary but well-specified json via GET-requests from the `/$value` endpoint.
+    - Allowing Changes to data via the PUT `submodel/submodel-elements/{{idShortPath}}` endpoint.
+
+Implementing the last point is tricky. It requires a very differentiated access control concept on the Supplier's side,
+differentiating WHO is allowed to write into WHAT part of the Submodel. Several approaches to implementation can be taken:
+
+#### 3.1. Access Control at the connector
+Connectors can prevent certain network participants from accessing the Submodel in the Supplier's backend. The EDC for
+example can intercept or rewrite components of an HTTP-request like headers, methods/verbs, bodies. Only extending a 
+data offer after preliminary checks is another option, implemented for instance in the EDC's Access Policy checks. A 
+combination of the two approaches can lead to fairly complete access control but may require fine-grained data-offers
+in the provider's catalog (i.e. very restricted EDC-Assets with a lot of proxy-parameters set `"false"`).
+
+#### 3.2. Access Control in the backend
+Depending on the backend's capabilities to fine-tune access control, it may be more beneficial to let the backend (like
+a Submodel Repository) decide if data will certain API-operations will be allowed. In this example, it would be possible
+to let a user update certain properties in a Submodel but only read others. This differentiation can't be made by the 
+connector as a connector will usually be oblivious to the data model and API structure of backend systems like a DTR or
+Submodel Repository.
+
+### 4. Updating an existing Twin
+
+| Scenario                                                                  | Participants                                                   | Assumptions                             | Links to Use-Cases |
+|---------------------------------------------------------------------------|----------------------------------------------------------------|-----------------------------------------|--------------------|
+| A new customer wants their customerPartId as specificAssetId on the twin. | - Data Provider (Supplier) <br/> - Data Provider (Third Party) | 1. Third Party knows an id of the asset | None yet           |
+
+An update to a twin is any change to the ShellDescriptor via the APIs of the AssetAdministrationShellServiceSpecification
```

Comment:
This use case is about changing the submodel that is registered in the Registry for this twin. So the registry information about this twin is not changed at all. Only if there is version information of the submodel contained in the registry (but this is optional)

Reply:
it's about arbitrary changes to a shell-descriptor. Adding submodels when onboarding use-cases, changing submodel-interfaces in migration scenario...